crates/algokit_transact_ffi/src/lib.rs

```diff
@@ -493,9 +493,25 @@ pub fn address_from_string(address: &str) -> Result<Address, AlgoKitTransactErro
         .map_err(|e| AlgoKitTransactError::EncodingError(e.to_string()))
 }
 
+/// Get the raw 32-byte transaction ID for an encoded transaction.
+#[ffi_func]
+pub fn get_transaction_raw_id(encoded_tx: &[u8]) -> Result<Vec<u8>, AlgoKitTransactError> {
+    let tx = algokit_transact::Transaction::decode(encoded_tx)?;
+    let raw_id = tx.raw_id()?;
```

Comment:
Calling `raw_id` re-encodes the transaction to msgpack before generating the hash.

I'm wondering if it's better to only support Transaction, which would be as below:

```rs
#[ffi_func]
pub fn get_transaction_raw_id(tx: Transaction) -> Result<Vec<u8>, AlgoKitTransactError> {
    let ctx: algokit_transact::Transaction = tx.try_into()?;
    let raw_id = ctx.raw_id()?;
    Ok(raw_id.to_vec())
}
```